pallets/nft/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+			Self::deposit_event(Event::NFTTokenBurned(sender, token.0, token.1));
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::destroy_class())]
+		pub fn destroy_class(origin: OriginFor<T>, class_id: ClassIdOf<T>) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			let class_info = orml_nft::Pallet::<T>::classes(class_id).ok_or(Error::<T>::ClassNotFound)?;
+			ensure!(sender == class_info.owner, Error::<T>::NotClassOwner);
+			ensure!(class_info.total_issuance == Zero::zero(), Error::<T>::NonZeroIssuance);
+			orml_nft::Pallet::<T>::destroy_class(&sender, class_id)?;
+			ClassItemPrice::<T>::remove(class_id);
+			Self::deposit_event(Event::NFTTokenClassDestroyed(sender, class_id));
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::buy_from_pool())]
+		pub fn buy_from_pool(origin: OriginFor<T>, token: (ClassIdOf<T>, TokenIdOf<T>)) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			let class_info = orml_nft::Pallet::<T>::classes(token.0).ok_or(Error::<T>::ClassNotFound)?;
+			let token_info = orml_nft::Pallet::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenNotFound)?;
+			let price = Self::class_item_price(token.0);
+			ensure!(class_info.owner == token_info.owner, Error::<T>::TokenAlreadyHasAnOwner);
+			ensure!(sender != token_info.owner, Error::<T>::CannotBuyOwnToken);
+			ensure!(!token_info.data.locked, Error::<T>::TokenLocked);
+
+			with_transaction_result(|| {
+				orml_nft::Pallet::<T>::transfer(&token_info.owner, &sender, token)?;
+				T::Currency::transfer(&sender, &token_info.owner, price, ExistenceRequirement::KeepAlive)?;
+				Self::deposit_event(Event::NFTBoughtFromPool(class_info.owner, sender, token.0, token.1));
+				Ok(())
+			})
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::sell_to_pool())]
+		pub fn sell_to_pool(origin: OriginFor<T>, token: (ClassIdOf<T>, TokenIdOf<T>)) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			let class_info = orml_nft::Pallet::<T>::classes(token.0).ok_or(Error::<T>::ClassNotFound)?;
+			let token_info = orml_nft::Pallet::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenNotFound)?;
+			let price = Self::class_item_price(token.0);
+			ensure!(class_info.owner != token_info.owner, Error::<T>::CannotSellPoolToken);
+			ensure!(sender == token_info.owner, Error::<T>::NotTokenOwner);
+			ensure!(!token_info.data.locked, Error::<T>::TokenLocked);
+
+			with_transaction_result(|| {
+				orml_nft::Pallet::<T>::transfer(&sender, &class_info.owner, token)?;
+				T::Currency::transfer(&class_info.owner, &sender, price, ExistenceRequirement::KeepAlive)?;
+				Self::deposit_event(Event::NFTSoldToPool(sender, class_info.owner, token.0, token.1));
+				Ok(())
+			})
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_finalize(now: T::BlockNumber) {
+			let bond = T::ClassBondAmount::get();
+			Self::unlock_bond(now, bond);
```

Comment:
This should be have some weight too, no ? So it has to be taken into account.

Reply:
added to create_class weight calculation